pallets/attesters/src/lib.rs

```diff
@@ -0,0 +1,1232 @@
+                .map(|(nominator, balance)| (nominator, balance))
+                .collect()
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        pub fn committee_size() -> usize {
+            T::CommitteeSize::get() as usize
+        }
+
+        fn update_sorted_nominated_attesters(
+            attester: &T::AccountId,
+            amount: BalanceOf<T>,
+        ) -> bool {
+            let mut all_indices_match = true;
+            SortedNominatedAttesters::<T>::mutate(|attesters| {
+                if let Some(index) = attesters.iter().position(|(a, _n)| a == attester) {
+                    let total_nomination = attesters[index].1 - amount;
+                    if total_nomination.is_zero() {
+                        attesters.remove(index);
+                    } else {
+                        attesters[index] = (attester.clone(), total_nomination);
+                    }
+
+                    // Sort the attesters by their nomination amount
+                    attesters.sort_by(|(_a1, n1), (_a2, n2)| n2.cmp(n1));
+
+                    // Keep only the top 32 attesters in the list
+                    attesters.truncate(32);
+                } else {
+                    log::warn!("Attester not found while updating sorted nominated attesters");
+                    all_indices_match = false
+                }
+            });
+            all_indices_match
+        }
+
+        pub fn do_nominate(
+            nominator: T::AccountId,
+            attester: T::AccountId,
+            amount: BalanceOf<T>,
+        ) -> DispatchResult {
+            // Check if nominator has enough balance
+            ensure!(
+                T::Currency::free_balance(&nominator) >= amount,
+                Error::<T>::NominatorNotEnoughBalance
+            );
+
+            let current_nomination =
+                Nominations::<T>::get(&attester, &nominator).unwrap_or(Zero::zero());
+
+            let new_nomination = current_nomination + amount;
+            Nominations::<T>::insert(&attester, &nominator, new_nomination);
+
+            // Update the sorted list of nominated attesters
+            SortedNominatedAttesters::<T>::try_mutate(|attesters| {
+                let total_nomination = Nominations::<T>::iter_prefix(&attester)
+                    .map(|(_, balance)| balance)
+                    .fold(Zero::zero(), |acc, balance| acc + balance);
+
+                if let Some(index) = attesters.iter().position(|(a, _n)| a == &attester) {
+                    // Update the existing nomination amount
+                    attesters[index] = (attester.clone(), total_nomination);
+                } else {
+                    // Add the new attester to the list
+                    attesters.push((attester.clone(), total_nomination));
+                }
+
+                // Sort the attesters by their nomination amount
+                attesters.sort_by(|(_a1, n1), (_a2, n2)| n2.cmp(n1));
```

Comment:
Shouldn't this only happen on shuffle?

Reply:
I'm updating sorted set after each nomination, but assigning new active set at shuffle

Reply:
if there's no un-nominations the sorting isn't necessary therefore